Add missing features from Travis CI test build

galjonsfigur opened the pull request:
Fixes #3254

- [x] This PR is for the `dev` branch rather than for `master`.
- [x] This PR is compliant with the [other contributing guidelines](https://github.com/nodemcu/nodemcu-firmware/blob/dev/CONTRIBUTING.md) as well (if not, please describe why).
- [x] I have thoroughly tested my contribution.
- [x] The code changes are reflected in the documentation at `docs/*`.

This PR adds all missing features to the Travis CI test build, at least all those I could find.

marcelstoer commented:
I support this. Can you please:
- fix the CI build
- unify the `sed` switches: ` -E -i.bak` vs. `-i.bak` vs. `-i` (haven't verified whether `-E` is actually necessary)
- fix indentation

galjonsfigur commented:
It should be fine now. I replaced strange `sed` parameters (`-E` option of `sed` isn't even documented anywhere...) and added small fixes as a separate commit.

marcelstoer commented:
> `-E` option of `sed` isn't even documented anywhere

I guess that depends on your environment. On my macOS it says

> -E      Interpret regular expressions as extended (modern) regular expressions rather than basic regular expressions (BRE's).
